pkg/rpcserver/runner.go

```diff
@@ -138,7 +138,7 @@ func (runner *Runner) ConnectionLoop() error {
 			infoc <- []byte("VM has crashed")
 		}
 	}()
-	for {
+	for runner.Alive() {
```

Comment:
If this is only about the sleeping case (VM check and tests), let's move this check to before the sleep.

Reply:
If the instance did not panic, but rather we've found some Report in the serial output and stopped fuzzing, I think we do want to avoid extra `source.Next()` calls -- why fetch them only to fail them a while after?

If it's extra sleep that's worrying, I can add an extra Alive() check before the sleep

Reply:
> I think we do want to avoid extra source.Next() calls -- why fetch them only to fail them a while after?

We won't fail them. If executor has not started executing them, then we will retry them w/o considering them as crashing.
But also it won't help much to prevent even that. The most time is spend in recv, program generation, and send. This check will help only if StopFuzzing called in the narrow window between we successfully send the previous program and called recv again, that's basically nanoseconds.
To handle stopFuzzing promptly and prevent wasted work, we would need to sprinkle Alive() calls on every line in pkg/fuzzer and prog packages. That does not make sense.

I care the least about the time.Sleep case, so I propose to move the Alive check there.

We also don't call StopFuzzing in any tests, so this shouldn't be the root cause of the test hangs we see.